# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: third_party/rust/tabs/src/sync/full_sync.rs
# Commit: 57b2cda7e0db
# Full Hash: 57b2cda7e0db8b07fcc1e09113ce26add178c68a
# Author: Sammy Khamis <skhamis@mozilla.com>
# Date: 2023-03-01 03:56:13
# Regressor Bug: 1818349
# File Overlap Count: 23
# Overlapping Files: third_party/rust/sync15/src/client/coll_update.rs, third_party/rust/sync15/src/client/mod.rs, third_party/rust/sync15/.cargo-checksum.json, third_party/rust/sync15/src/client/sync.rs, Cargo.toml
# Description:
#   Bug 1818349 Part 1: Vendor new version of application-services r=markh
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D171181
# ==============================================================================

diff -r eaf9f12a1481 -r 57b2cda7e0db third_party/rust/tabs/src/sync/full_sync.rs
--- a/third_party/rust/tabs/src/sync/full_sync.rs	Tue Feb 28 04:22:35 2023 +0000
+++ b/third_party/rust/tabs/src/sync/full_sync.rs	Tue Feb 28 05:14:29 2023 +0000
@@ -2,7 +2,7 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
-use crate::{sync::engine::TabsSyncImpl, ApiResult, Result, TabsEngine, TabsStore};
+use crate::{sync::engine::TabsSyncImpl, ApiResult, TabsEngine, TabsStore};
 use error_support::handle_error;
 use interrupt_support::NeverInterrupts;
 use std::sync::Arc;
@@ -11,7 +11,7 @@
 use sync15::KeyBundle;
 
 impl TabsStore {
-    #[handle_error]
+    #[handle_error(crate::Error)]
     pub fn reset(self: Arc<Self>) -> ApiResult<()> {
         let mut sync_impl = TabsSyncImpl::new(Arc::clone(&self));
         sync_impl.reset(&EngineSyncAssociation::Disconnected)?;
@@ -19,7 +19,7 @@
     }
 
     /// A convenience wrapper around sync_multiple.
-    #[handle_error]
+    #[handle_error(crate::Error)]
     pub fn sync(
         self: Arc<Self>,
         key_id: String,
